sdk/storage/azure-storage-common/src/file_io.cpp

```diff
@@ -22,35 +22,42 @@
 #include <windows.h>
 #endif
 
-#include <codecvt>
 #include <limits>
-#include <locale>
 #include <stdexcept>
 
 namespace Azure { namespace Storage { namespace _internal {
 
 #if defined(AZ_PLATFORM_WINDOWS)
   FileReader::FileReader(const std::string& filename)
   {
+    int sizeNeeded
+        = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, filename.data(), -1, nullptr, 0);
+    if (sizeNeeded == 0)
+    {
+      throw std::runtime_error("invalid filename");
+    }
+    std::wstring filenameW(sizeNeeded, L'\0');
```

Comment:
`sizeNeeded` includes the terminating `L'\0'` when the `cbMultiByte` argument to `MultiByteToWideChar` is `-1` as it is currently. Consequently, this string will be one character longer than necessary and have two `L'\0'` characters at the end. If you don't apply my suggestion above, I'd change this to `filenameW(sizeNeeded - 1, L'\0')`.

Reply:
Hi @CaseyCarter , I tested with code

```C++
  std::string str = "abc";
  int sizeNeeded = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, str.data(), -1, nullptr, 0);
```

if it's `-1`, the return value is `4`. If it's `str.length()`, the return value is `3`.

So if I apply your suggestion above, this line should be `std::wstring filenameW(sizeNeeded + 1, L'\0')`. Is that correct?